webpack/assets/javascripts/react_app/components/componentRegistry.js

```diff
@@ -36,15 +36,17 @@ import LoginPage from './LoginPage';
 import ExternalLogout from './ExternalLogout';
 import Slot from './common/Slot';
 
+window.tfm_component_registry = window.tfm_component_registry || {};
+
 const componentRegistry = {
-  registry: {},
+  registry: window.tfm_component_registry,
 
   register({ name = null, type = null, store = true, data = true }) {
     if (!name || !type) {
       throw new Error('Component name or type is missing');
     }
     if (this.registry[name]) {
-      throw new Error(`Component name already taken: ${name}`);
+      return this.registry;
```

Comment:
Can we add a warning before returning?

It can turn to:
```js
if (this.registry[name]) {
  console.warning(`Component name already taken: ${name}`);
} else {
  this.registry[name] = { type, store, data };
}
```

Reply:
Is there a better way to describe a component? In many languages you can compare the memory address as well to see if it's the same instance or actually a different one. That might also be interesting to know because it can indicate whether the initializer is called twice with the same component or whether the code is actually duplicated

Reply:
Well, you can do `type === this.registry[name].type` and it would only compare the pointers.

Reply:
we could, but the question is does it matter? we might have a plugin create a different instance of a component and try to register it. In that case the name would be the same but the memory address would be different. I think logging is enough for us to see the message and try to figure out what happened, and we can improve on it in the future if needed but right now i want to first get to a state where we can actually build plugins again.